pages/_app.js

```diff
@@ -0,0 +1,43 @@
+import {
+  HydrationBoundary,
+  QueryClient,
+  QueryClientProvider,
+} from "@tanstack/react-query";
+import { ReactQueryDevtools } from "@tanstack/react-query-devtools";
+import Header from "@/components/Header/Header";
+import "@/styles/main.scss";
+import Head from "next/head";
+import Footer from "@/components/Footer/Footer";
+import Main from "@/components/Main/Main";
+import React from "react";
+
+export default function App({ Component, pageProps }) {
+  const [queryClient] = React.useState(
+    () =>
+      new QueryClient({
+        defaultOptions: {
+          queries: {
+            staleTime: 60 * 1000,
+          },
+        },
+      })
+  );
```

Comment:
https://tanstack.com/query/latest/docs/framework/react/quick-start

보통 
```
const queryClient = new QueryClient()

function App() {
  return (
    // Provide the client to your App
    <QueryClientProvider client={queryClient}>
      <Todos />
    </QueryClientProvider>
  )
}
```
이렇게 세팅하는데 혹시 이렇게 작성한 이유가 있나요?

Reply:
https://tanstack.com/query/latest/docs/framework/react/guides/ssr
next page 라우트 기반에서 ssr이랑 hydration 찾다가 공식문서 이부분을 봐서 저렇게 따라 했습니다.
  // With SSR, we usually want to set some default staleTime
  // above 0 to avoid refetching immediately on the client
  
  요렇게 써있어서 staleTime을 예제처럼 넣어줬습니담.